### PR TITLE
EOS-15690 : S3 compatibility test scripts

### PR DESCRIPTION
**Pull request by gowthamchinna**
Signed-off-by: Gowthaman Chinnathambi <gowthaman.chinnathambi@seagate.com>

### CODE REVIEW

File: scripts/compatibility-test/run-test.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+
+_help_function(){
```
Comment (nileshgovande) [inline scripts/compatibility-test/run-test.sh:22]: Pls add (at least) one liner as function header comment

**Review by nileshgovande — APPROVED** [review]

File: scripts/compatibility-test/run-test.sh (hunk 1 above)
Comment (gowthamchinna) [reply scripts/compatibility-test/run-test.sh:22]: Incorporated the suggestion

File: scripts/compatibility-test/readme.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+   1. S3Server should be up and running 
+   
+Steps:
+
+1. Navigate to the compatibility-test script folder
+   ```
+      cd cortx-s3server/scripts/compatibility-test
+   ```
+
+2. create s3test.conf for the corresponding integration. Refer sample template for ceph and plunk integrations in the respective folders. Two s3 account credentials need to be updated in the configurations
```
Comment (sachitanands) [inline scripts/compatibility-test/readme.md:14]: minor spelling mistake pls correct it as "splunk "

File: scripts/compatibility-test/readme.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      ```
+      - Run Splunk tests
+      
+      ```
+         sh run-test.sh -c splunk/splunk.conf -i splunk
+      ```
+      
+      - Run V2 or V4 test only 
+      ```
+         sh run-test.sh -c conf_location -i integration type (splunk | ceph) -t v4
```
Comment (sachitanands) [inline scripts/compatibility-test/readme.md:31]: Can be documented as 
"sh run-test.sh -c conf_location -i integration_type (i.e splunk | ceph) -t v4"

File: scripts/compatibility-test/run-test.sh [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+      sleep 30s
+fi
+
+# Executes V4 tests
+if [ "${S3_TEST_VERSION}" = "all" -o "${S3_TEST_VERSION}" = "v4" ]; then
+      
+      echo "[ ${S3_INTEGRATION_TYPE} ] : 2. Running V4 tests"
+      
+      # Running test with aws v2signature
```
Comment (sachitanands) [inline scripts/compatibility-test/run-test.sh:129]: comment msg should be "v4 signature"

File: scripts/compatibility-test/run-test.sh [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+      _execute_compatability_tets "${S3_TEST_DATA_V4}"
+
+      sleep 30s
+fi
+
+# Disabling virtual env
+deactivate
+
+echo "[ ${S3_INTEGRATION_TYPE} ] : compatibility test completed!!!"
```
Comment (sachitanands) [inline scripts/compatibility-test/run-test.sh:140]: please add new line at the end of these script file

**Review by sachitanands — COMMENTED** [review]
Added few minor comments ..pls address them and rest of the code looks good

File: scripts/compatibility-test/readme.md (hunk 2 above)
Comment (gowthamchinna) [reply scripts/compatibility-test/readme.md]: Fixed the typo

File: scripts/compatibility-test/readme.md (hunk 3 above)
Comment (gowthamchinna) [reply scripts/compatibility-test/readme.md]: Updated the sample cmd as per suggestion

File: scripts/compatibility-test/run-test.sh (hunk 4 above)
Comment (gowthamchinna) [reply scripts/compatibility-test/run-test.sh]: Fixed the wrong comments

File: scripts/compatibility-test/run-test.sh (hunk 5 above)
Comment (gowthamchinna) [reply scripts/compatibility-test/run-test.sh]: Added new line

**Review by sachitanands — APPROVED** [review]

**Pull request merged**